my_enumerable.rb

```diff
@@ -0,0 +1,210 @@
+
+# rubocop:disable all
+module Enumerable
+  def my_each
+    for item in self
+      if block_given?
+        yield(item)
+      else
+        item
+      end
+    end
+  end
+
+  def my_each_with_index
+    index=0
+    for item in self
+      if block_given?
+      yield(item, index)
+      else
+        item
+      end
+      index+=1
+    end
+  end
+
+  def my_select
+     result =[]
+    for item in self
+      if block_given?
+        result.push(item) if yield(item) == true
+      else
+        item
+      end
+   end
+   result
+ end
```

Comment:
- [ ] my_select

returns an Enumerator if no block is given:
`p [1, 2, 3].my_select` returns `#<Enumerator>`